packages/shared/src/components/tooltips/InteractivePopup.tsx

```diff
@@ -42,29 +44,81 @@ const positionClass: Record<InteractivePopupPosition, string> = {
   leftCenter: classNames(leftClass, centerClassY),
   leftEnd: classNames(leftClass, endClass),
   screen: 'inset-0 w-screen h-screen',
-  mainFeedLeftEnd: classNames(mainFeedLeftClass, mainFeedEndClass),
 };
 
+const leftPositions = [
+  InteractivePopupPosition.LeftEnd,
+  InteractivePopupPosition.LeftCenter,
+  InteractivePopupPosition.LeftEnd,
+];
+
+const WAIT_FOR_FULL_RENDER = 10;
+
 function InteractivePopup({
   children,
   className,
   position = InteractivePopupPosition.Center,
+  closeOutsideClick,
+  onClose,
   ...props
 }: InteractivePopupProps): ReactElement {
-  const classes = positionClass[position];
+  const container = useRef<HTMLDivElement>();
+  const onCloseRef = useRef(onClose);
+  const { sidebarRendered } = useSidebarRendered();
+  const { sidebarExpanded } = useSettingsContext();
+  const finalPosition = sidebarRendered
+    ? position
+    : InteractivePopupPosition.Center;
+  const classes = positionClass[finalPosition];
+
+  useEffect(() => {
+    if (!closeOutsideClick || !onCloseRef?.current) {
+      return null;
+    }
+
+    const onClickAnywhere = (e: MouseEvent) => {
+      if (container.current && !container.current.contains(e.target as Node)) {
+        onCloseRef.current(e);
+      }
+    };
+
+    // The reason we wait for a few ms is, when the user clicks the triggering element to open this,
+    // it is considered outside the core container. Hence, triggering the close here.
+    setTimeout(() => {
+      globalThis?.document.addEventListener('click', onClickAnywhere);
+    }, WAIT_FOR_FULL_RENDER);
```

Comment:
Can't we evaluate it by checking if container.current exist for instance?
It feels a bit like overkill.

If really no other way (which I doubt) let's use useDebounce rather.

Reply:
Let me know if you have other alternatives to that.

With regards to using debouce, compared to now I think it became more complicated than it needs to be with all the dependency management as we have to be very strict since this will add numerous listeners if handled incorrectly. Wdyt?
![Screenshot 2023-11-08 at 9 11 16 PM](https://github.com/dailydotdev/apps/assets/13744167/36ead1c9-4bf0-4f10-ac9f-69d293d2071b)

Reply:
To me hooks are there for exactly this reason, not re-use same logic, and handle all cleanup internally.
I think we also have a even listener hook maybe that one icw is better?

Not sure, @capJavert what do you think?

Reply:
Sorry I am not sure why the above needs a timeout in the first place, probably need some more context. 🙏 

What I do see is that with current setup we could get events leak because useEffect cleanup could fire before `setTimeout` executes in a first place. This is something we need to fix definitely. We can do that by using `clearTimeout` in cleanup function.

Reply:
@capJavert good catch on clearing the timeout.

@rebelchris I agree we should utilize the hooks whenever possible, but I think it is better to avoid them wherever we can as it is more performant without the additional overhead behind the scenes with checking the dependencies.

Reply:
@sshanzel That sounds very contradicting 😂
Maybe we can sync on that tomorrow quickly?

Reply:
Let me see how we can modify the useWindowEvents to support timeouts.

Reply:
> @sshanzel That sounds very contradicting 😂
Maybe we can sync on that tomorrow quickly?

Well, I think that is me most of the time 😂 
Sure, let's sync tomorrow.